src/com/namelessmc/java_api/NamelessAPI.java

```diff
@@ -117,18 +123,22 @@ public void checkWebAPIConnection() throws NamelessException {
 	public List<Announcement> getAnnouncements(final NamelessUser user) throws NamelessException {
 		final JsonObject response = this.requests.get(Action.GET_ANNOUNCEMENTS, "id", user.getId());
 
-		return jsonArrayToList(response.get("announcements").getAsJsonArray());
+		return getAnnouncements(response);
 	}
 
-	private List<Announcement> jsonArrayToList(JsonArray array) {
-		final List<Announcement> announcements = new ArrayList<>();
-		array.forEach((element) -> {
+	private List<Announcement> getAnnouncements(JsonObject response) {
+		return jsonArrayToList(response.get("announcements").getAsJsonArray(), element -> {
 			final JsonObject announcementJson = element.getAsJsonObject();
 			final String content = announcementJson.get("content").getAsString();
 			final String[] display = jsonToArray(announcementJson.get("display").getAsJsonArray());
 			final String[] permissions = jsonToArray(announcementJson.get("permissions").getAsJsonArray());
-			announcements.add(new Announcement(content, display, permissions));
+			return new Announcement(content, display, permissions);
 		});
+	}
+
+	private <T> List<T> jsonArrayToList(JsonArray array, Function<JsonElement, T> elementSupplier) {
+		final List<T> announcements = new ArrayList<>();
+		array.forEach(element -> announcements.add(elementSupplier.apply(element)));
 		return announcements;
```

Comment:
alternatively: `return StreamSupport.stream(array.spliterator(), false).map(elementSupplier::apply).collect(Collectors.toUnmodifiableList())`

Reply:
I could use `Collectors.toList()`, `Collectors.toUnmodifiableList()` is documented as Java 10 and above. In the pom.xml is defined, that Java 1.8 should be used for compile

Reply:
yes, that's fine